app/models/proposal.rb

```diff
@@ -69,11 +69,12 @@ def delegate?(user)
   end
 
   def approval_for(user)
-    # TODO convert to SQL
-    self.approvals.find do |approval|
-      approver = approval.user
-      approver == user || approver.outgoing_delegates.exists?(assignee_id: user.id)
-    end
+    where_clause = <<-SQL
+      user_id = :user_id
+      OR user_id IN (SELECT assigner_id FROM approval_delegates WHERE assignee_id = :user_id)
+      OR user_id IN (SELECT assignee_id FROM approval_delegates WHERE assigner_id = :user_id)
+    SQL
+    self.approvals.where(where_clause, user_id: user.id).first
```

Comment:
Yep. Previously we'd only needed to look up the relationship when a delegate was trying to approve. Here we also need to look the other way: is there a delegate who has already acted?